internal/gateway/backend/lease_actions.go

```diff
@@ -153,30 +163,38 @@ func (s *Services) CommitLease(ctx context.Context, tokenStr, oldRootHash, newRo
 	leasePath, lease, err := s.Leases.GetLease(ctx, tokenStr)
 	if err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
 
 	if err := CheckToken(tokenStr, lease.Token.Secret); err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
 
 	repository, _, err := gw.SplitLeasePath(leasePath)
 	if err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
 	}
+	var finalRev uint64
 	if err := s.Leases.WithLock(ctx, repository, func() error {
-		return s.Pool.CommitLease(ctx, leasePath, oldRootHash, newRootHash, tag)
+		var err error
+		finalRev, err = s.Pool.CommitLease(ctx, leasePath, oldRootHash, newRootHash, tag)
+		return err
 	}); err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
+	}
+
+	plotsErr := s.StatsMgr.UploadStatsPlots(repository)
+	if plotsErr != nil {
+		gw.LogC(ctx, "actions", gw.LogError).Msgf(plotsErr.Error())
 	}
 
 	if err := s.Leases.CancelLease(ctx, tokenStr); err != nil {
 		outcome = err.Error()
-		return err
+		return 0, err
```

Comment:
At this point we have commit the revision number `finalRev`, haven't we?

Should we return that even if the cancellation of the lease failed?

It breaks the invariant that `finalRev == 0` => some error happened, but it is more correct.

Reply:
Yeah, that's better. I wouldn't care about any invariant. We have 'err' for that